src/Spice86.Core/Emulator/Memory/IMemory.cs

```diff
@@ -0,0 +1,169 @@
+    public void LoadData(uint address, ushort[] data);
+
+    /// <summary>
+    ///     Load data from a word array into memory.
+    /// </summary>
+    /// <param name="address">The memory address to start writing</param>
+    /// <param name="data">The array of words to write</param>
+    /// <param name="length">How many words to read from the byte array</param>
+    public void LoadData(uint address, ushort[] data, int length);
+
+    /// <summary>
+    ///     Copy bytes from one memory address to another.
+    /// </summary>
+    /// <param name="sourceAddress">The address in memory to start reading from</param>
+    /// <param name="destinationAddress">The address in memory to start writing to</param>
+    /// <param name="length">How many bytes to copy</param>
+    public void MemCopy(uint sourceAddress, uint destinationAddress, uint length);
+
+    /// <summary>
+    ///     Fill a range of memory with a value.
+    /// </summary>
+    /// <param name="address">The memory address to start writing to</param>
+    /// <param name="value">The byte value to write</param>
+    /// <param name="amount">How many times to write the value</param>
+    public void Memset8(uint address, byte value, uint amount);
+
+    /// <summary>
+    ///     Fill a range of memory with a value.
+    /// </summary>
+    /// <param name="address">The memory address to start writing to</param>
+    /// <param name="value">The ushort value to write</param>
+    /// <param name="amount">How many times to write the value</param>
+    public void Memset16(uint address, ushort value, uint amount);
+
+    /// <summary>
+    ///     Find the address of a value in memory.
+    /// </summary>
+    /// <param name="address">The address in memory to start the search from</param>
+    /// <param name="len">The maximum amount of memory to search</param>
+    /// <param name="value">The sequence of bytes to search for</param>
+    /// <returns>The address of the first occurence of the specified sequence of bytes, or null if not found.</returns>
+    uint? SearchValue(uint address, int len, IList<byte> value);
+
+    /// <summary>
+    ///     Enable or disable a memory breakpoint.
+    /// </summary>
+    /// <param name="breakPoint">The breakpoint to enable or disable</param>
+    /// <param name="on">true to enable a breakpoint, false to disable it</param>
+    /// <exception cref="NotSupportedException"></exception>
+    void ToggleBreakPoint(BreakPoint breakPoint, bool on);
+    
+    /// <summary>
+    ///     Allows write breakpoints to access the byte being written before it actually is.
+    /// </summary>
+    byte CurrentlyWritingByte {
+        get;
+        set;
+    }
+
+    /// <summary>
+    ///     The number of bytes in the memory map.
+    /// </summary>
+    int Size { get; }
+
+    /// <summary>
+    ///     Allows indexed byte access to the memory map.
+    /// </summary>
+    public UInt8Indexer UInt8 { get; }
```

Comment:
Is there a difference between `IMemory.UInt8` and `IIndexed.UInt8`?
Perhaps this definition should be removed from one of them?

Reply:
I have to move things that are coming from Indexable to a new interface for unit tests, maybe a future IIndexable.

Right now there are type casts of _machine.Memory (which is declared as an instance of IMemory) as an Indexable and I don't like it.

Reply:
Removed duplicated UInt8 and such in IMemory.cs